extension/src/frequency.rs

```diff
@@ -712,6 +1171,64 @@ pub fn min_frequency(
     0.
 }
 
+#[pg_extern(immutable, parallel_safe, name="max_frequency", schema = "toolkit_experimental")]
+pub fn max_bigint_frequency(
+    agg: SpaceSavingBigIntAggregate<'_>,
+    value: i64,
+) -> f64 {
+    for (idx, datum) in agg.datums.iter().enumerate() {
+        if value == datum as i64 {
```

Comment:
I've run out of time, but the same approach can be applied to max and min frequency functions: implement the algorithm in generic function parametrized over data type and conversion function.